### PR TITLE
feat: readme parser

### PR DESCRIPTION
**Pull request by r4zendev**

### CODE REVIEW

File: readme-parser/parse.test.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+f_github-link: -
+f_vs-code-link: -
+f_codemod-studio-link: -
+f_cli-command: -
+f_framework: -
+f_applicability-criteria: MSW >= 1.0.0
+f_verified-codemod: true
+f_author: -
+layout: "[automations].html"
+slug: -
```
Comment (grzpab) [inline readme-parser/parse.test.ts:171]: do we need to provide it? we can slugify the title I think

File: readme-parser/parse.test.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  return res(ctx.json({ firstName: 'John' }));
+});
+\`\`\`
+After
+\`\`\`ts
+http.get<any, ReqBodyType>('/resource', (req, res, ctx) => {
+  return res(ctx.json({ firstName: 'John' }));
+});
+\`\`\`
+f_github-link: -
```
Comment (grzpab) [inline readme-parser/parse.test.ts:162]: can we generate these links as well?

File: readme-parser/parse.test.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+f_vs-code-link: -
+f_codemod-studio-link: -
```
Comment (grzpab) [inline readme-parser/parse.test.ts:164]: the codemod hash digest comes from `createHash('ripemd160').update(codemod.name).digest('base64url)`

File: readme-parser/parse.ts [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+					`\`\`\`${rc.lang}\n${rc.value}\n\`\`\`${delimiter}`,
+				);
+			} else {
+				textParts.push(`${rc.value}${delimiter}`);
+			}
+		}
+	}
+
+	// Trim last el to remove delimiter
+	const lastElement = textParts[textParts.length - 1];
```
Comment (grzpab) [inline readme-parser/parse.ts:145]: I wonder if `.at(-1)` wouldn't be easier to read here?

File: readme-parser/parse.ts [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+	if (lastElement) {
+		const lastSymbol = lastElement[lastElement.length - 1];
```
Comment (grzpab) [inline readme-parser/parse.ts:147]: can it be regexed?

File: readme-parser/parse.test.ts (hunk 2 above)
Comment (r4zendev) [reply readme-parser/parse.test.ts:162]: generated in the next PR

File: readme-parser/parse.test.ts (hunk 1 above)
Comment (r4zendev) [reply readme-parser/parse.test.ts:171]: generated in the next PR

File: readme-parser/parse.ts (hunk 5 above)
Comment (r4zendev) [reply readme-parser/parse.ts]: i would argue that this is way easier to read than what i've replaced it with, but it works for me

File: readme-parser/parse.test.ts (hunk 3 above)
Comment (r4zendev) [reply readme-parser/parse.test.ts:164]: fixed in #538

**Review by grzpab — APPROVED** [review]

**Pull request merged**